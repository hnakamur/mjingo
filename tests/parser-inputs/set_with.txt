Basic:
{{ foo }}
{% set foo = "new value" %}
{{ foo }}
{% with %}
  {% set foo = "new value 2" %}
  {{ foo }}
{% endwith %}
{{ foo }}
