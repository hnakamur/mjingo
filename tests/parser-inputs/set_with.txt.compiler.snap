(
    [
        00000 | EmitRaw("Basic:\n")  [line 1],
        00001 | Lookup("foo")  [line 2],
        00002 | Emit,
        00003 | EmitRaw("\n"),
        00004 | LoadConst("new value")  [line 3],
        00005 | StoreLocal("foo"),
        00006 | EmitRaw("\n"),
        00007 | Lookup("foo")  [line 4],
        00008 | Emit,
        00009 | EmitRaw("\n"),
        00010 | PushWith  [line 5],
        00011 | EmitRaw("\n  "),
        00012 | LoadConst("new value 2")  [line 6],
        00013 | StoreLocal("foo"),
        00014 | EmitRaw("\n  "),
        00015 | Lookup("foo")  [line 7],
        00016 | Emit,
        00017 | EmitRaw("\n"),
        00018 | PopFrame,
        00019 | EmitRaw("\n")  [line 8],
        00020 | Lookup("foo")  [line 9],
        00021 | Emit,
    ],
    {},
)
